packages/grid/_modules_/grid/hooks/utils/useGridApiContext.ts

```diff
@@ -14,5 +18,5 @@ export function useGridApiContext() {
     );
   }
 
-  return apiRef;
+  return apiRef as any;
```

Comment:
Without as `as`, Typescript complains that the state value can have a different type.

But the following works and is cleaner:

```suggestion
  return apiRef as GridApiRef<GridApi>;
```

Reply:
Interesting!

I think this is the reason:
![Screenshot 2022-01-26 at 12 46 14](https://user-images.githubusercontent.com/13808724/151157929-061ebc29-227d-4ae2-b9dd-85cc64f06982.png)
 
 `GridApiContext` uses `GridApiRef` which uses `GridApiCommunity` by default.
 
I've found this solution:

```tsx
export function useGridApiContext<GridApi extends GridApiCommon = GridApiCommunity>() {
  // override GridApiContext type here
  const apiRef = React.useContext(GridApiContext as React.Context<GridApiRef<GridApi> | undefined>);

  if (apiRef === undefined) {
    throw new Error(
      [
        'MUI: Could not find the data grid context.',
        'It looks like you rendered your component outside of a DataGrid or DataGridPro parent component.',
        'This can also happen if you are bundling multiple versions of the data grid.',
      ].join('\n'),
    );
  }

  return apiRef;
}
```

Which returns proper type for returned apiRef:
![Screenshot 2022-01-26 at 12 50 43](https://user-images.githubusercontent.com/13808724/151158412-b01d3ab2-b058-4857-b168-79421a9b8350.png)

Not sure if it's any better than your solution.
What do you think?

Reply:
It moves the `as` somewhere else, I don't think it's very different in the end :+1: